rsts/deployment/devops.rst

```diff
@@ -0,0 +1,41 @@
+.. _devops-ideal-flow:
+
+Ideal Flow for DevOps
+---------------------
+
+Flyte could be useful for many use cases, be it model training, data processing, ELT/ETL, or bioinformatics.
+When workflows are built and deployed, be it in any domain, we may need automation to reduce human-in-the-loop to some extent;
+essentially, to automate serializing and registering workflows, creating docker containers, etc.
+
+Before diving into an example use case that explains how DevOps could power Flyte pipelines, let's look at some DevOps-style features Flyte ships with.
+
+- Every DAG is versioned with Git commit's SHA, which is modifiable
+- Tasks and workflows are immutable
+- In the case of rapid iteration of code with no dependency modification, :ref:`deployment-fast-registration` can be used
+- Multi-tenancy support through projects and domains
+- Executions can be monitored using logs on :ref:`Flyte UI <ui>`
+
+Lyft's Use Case
+===============
+
+Flyte powers more than 1 million model training and data processing executions per month at Lyft.
+DevOps automation has been made part of Flyte pipelines to handle such a massive number of executions.
+
+- Every time a new PR is created:
+
+  - A Docker container is built from the PR
+  - Tasks and workflows are registered with Flyte
+- If dependencies are modified, PR securely builds the container again
+- For just code changes, fast registration is used
```

Comment:
Can be done from laptop, users are not permitted to push docker containers from local machines